community/kernel-api/src/main/java/org/neo4j/internal/kernel/api/ExplicitIndexWrite.java

```diff
@@ -22,24 +22,131 @@
 
 /**
  * Operations for creating and modifying explicit indexes.
  */
 public interface ExplicitIndexWrite
 {
+    /**
+     * Adds node to explicit index.
+     *
+     * @param indexName The name of the index
+     * @param node The id of the node to add
+     * @param key The key to associate with the node
+     * @param value The value to associate with the node an key
+     * @throws ExplicitIndexNotFoundKernelException If there is no explicit index with the given name
+     */
+    void nodeAddToExplicitIndex( String indexName, long node, String key, Object value )
+            throws KernelException;
+
+    /**
+     * Removes a node from an explicit index
+     *
+     * @param indexName The name of the index
+     * @param node The id of the node to remove
+     * @param key The key to associate with the node
+     * @param value The value to associate with the node and key
+     * @throws ExplicitIndexNotFoundKernelException If there is no explicit index with the given name
+     */
+    void nodeRemoveFromExplicitIndex( String indexName, long node, String key, Object value )
+            throws ExplicitIndexNotFoundKernelException;
+
+    /**
+     * Removes a node from an explicit index
+     *
+     * @param indexName The name of the index
+     * @param node The id of the node to remove
+     * @param key The key to associate with the node
+     * @throws ExplicitIndexNotFoundKernelException If there is no explicit index with the given name
+     */
+    void nodeRemoveFromExplicitIndex( String indexName, long node, String key ) throws
+            ExplicitIndexNotFoundKernelException;
+
     /**
      * Removes a given node from an explicit index
      *
      * @param indexName The name of the index from which the node is to be removed.
      * @param node The node id of the node to remove
+     * @throws ExplicitIndexNotFoundKernelException If there is no explicit index with the given name
+     */
+    void nodeRemoveFromExplicitIndex( String indexName, long node ) throws ExplicitIndexNotFoundKernelException;
+
+    /**
+     * Adds relationship to explicit index.
+     *
+     * @param indexName The name of the index
+     * @param relationship The id of the relationship to add
+     * @param key The key to associate with the relationship
+     * @param value The value to associate with the relationship and key
+     * @throws ExplicitIndexNotFoundKernelException If there is no explicit index with the given name
+     */
+    void relationshipAddToExplicitIndex( String indexName, long relationship, String key, Object value )
+            throws KernelException;
+
+    /**
+     * Removes relationship from explicit index.
+     *
+     * @param indexName The name of the index
+     * @param relationship The id of the relationship to remove
+     * @param key The key to associate with the relationship
+     * @param value The value to associate with the relationship and key
```

Comment:
@Param key The key ~to associate~ associated with the relationship
@Param value The value ~to associate~ associated with the relationship and key